File: C1.UWP.FlexGrid/CS/EMenus/Data/CustomTemplateSelector.cs
using C1.Xaml;
using Windows.UI.Xaml;

namespace Grapecity.C1_EMenus
{
    public class CustomTemplateSelector : C1DataTemplateSelector
    {
        protected override DataTemplate SelectTemplateCore(object item, DependencyObject container)
        {
            if (item is Category)
            {                
                return Resources["CategoryTemplate"] as DataTemplate;
            }
            if (item is SubCategory)
            {
                return Resources["SubCategoryTemplate"] as DataTemplate;
            }
            return null;
        }
    }
}

File: C1.UWP.FlexGrid/VB/EMenus/Data/CustomTemplateSelector.vb
Imports C1.Xaml

Public Class CustomTemplateSelector
    Inherits C1DataTemplateSelector
    Protected Overrides Function SelectTemplateCore(item As Object, container As DependencyObject) As DataTemplate
        If TypeOf item Is Category Then
            Return TryCast(Resources("CategoryTemplate"), DataTemplate)
        End If
        If TypeOf item Is SubCategory Then
            Return TryCast(Resources("SubCategoryTemplate"), DataTemplate)
        End If
        Return Nothing
    End Function
End Class
